wkcuber/api/Layer.py

```diff
@@ -99,20 +118,157 @@ def set_bounding_box_size(self, size: Tuple[int, int, int]) -> None:
         for _, mag in self.mags.items():
             mag.view.size = size
 
-    def setup_mag(self, mag: str) -> None:
+    def _initialize_mag_from_other_mag(
+        self, new_mag_name: Union[str, Mag], other_mag: MagDataset, compress: bool
+    ) -> MagDataset:
+        raise NotImplemented
+
+    def downsample(
+        self,
+        from_mag: Mag,
+        max_mag: Mag,
+        interpolation_mode: str,
+        compress: bool,
+        scale: Tuple[float, float, float] = None,
+        buffer_edge_len: int = None,
+        args: Namespace = None,
+    ) -> None:
+        # if 'scale' is set, the data gets downsampled anisotropic
```

Comment:
What do you mean with "scale of the layer"?
Let say the layer has 2 existing magnifications ("1", "2"). The user can now downsample the data anisotropic (`from_mag="2"`, ..., `scale="1-1-2"`). This would create `4-4-2`-

Reply:
I think, the `scale` argument here is the scale which is persisted in the datasource-properties.json. So, for example it could be `[11.24, 11.24, 24]` or something equally odd/fractional as it's referring to the nm size of a voxel. Since this is stored in the dataset's properties, the layer could probably look this up easily via its dataset.

Speaking closer to the code: Have a look at the following in downsampling.py:

```python
    if anisotropic:
        if scale is None:
            try:
                scale = read_datasource_properties(path)["scale"]
            except Exception as exc:
                logging.error(
                    "Could not get the scale from the datasource-properties.json. Probably your path is wrong. "
                    "If you do not provide the layer_name or from_mag, they need to be included in the path."
                    "(e.g. dataset/color/1). Otherwise the path should just point at the dataset directory."
                    "the path: %s",
                    path,
                )

    WKDataset(path).get_layer(layer_name).downsample(
        from_mag=from_mag,
        max_mag=max_mag,
        interpolation_mode=interpolation_mode,
        compress=compress,
        scale=scale,
        buffer_edge_len=buffer_edge_len,
        args=args,
    )
```

Instead of extracting the scale from the outside, the actual `downsample` method could do so. It would only need the boolean information `anisotropic` (or `use_scale`; I don't have a strong opinion here). Otherwise, if some other code wants to downsample a layer anisotropically, they would need to look up the scale manually.

Reply:
Ok, very interesting. I like your idea and will implement it. I just always assumed that you could specify any scale as parameter (independent of the the scale in the properties)

Reply:
> I just always assumed that you could specify any scale as parameter (independent of the the scale in the properties)

Yes, this might have been the case, but I'd argue that this wasn't really intended during the design of the old interface. Instead, it can be quite error prone. If one wants to create a custom mag hierarchy, one has to do this manually in my opinion.